pilot/pkg/xds/pcds.go

```diff
@@ -0,0 +1,52 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xds
+
+import (
+	mesh "istio.io/api/mesh/v1alpha1"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/util/gogo"
+)
+
+// PcdsGenerator generates ECDS configuration.
+type PcdsGenerator struct {
+	Server *DiscoveryServer
+}
+
+var _ model.XdsResourceGenerator = &PcdsGenerator{}
+
+func pcdsNeedsPush(req *model.PushRequest) bool {
+	if req == nil {
+		return true
+	}
+	// If none set, we will always push
+	if len(req.ConfigsUpdated) == 0 {
+		return true
+	}
+	return false
+}
+
+// Generate returns ECDS resources for a given proxy.
+func (e *PcdsGenerator) Generate(proxy *model.Proxy, push *model.PushContext, w *model.WatchedResource, req *model.PushRequest) (model.Resources, error) {
+	if !pcdsNeedsPush(req) {
+		return nil, nil
+	}
+
+	pc := &mesh.ProxyConfig{
+		// TODO update to fetch the real cert, and use the new API to store it.
```

Comment:
Pretty sure we have the code that loads the cert from Secret and stores it somewhere.